Franklin/App/AppController/AppController.swift

```diff
@@ -346,37 +346,54 @@ public class AppController {
         
         do {
             try wallet.add(token: buff,
-                           network: Web3Network(id: 100, name: "xDai"))
+                           network: XDaiNetwork())
         } catch let error {
             throw error
         }
         self.userDefaultKeys.setBuffAdded(for: wallet)
     }
     
+    public func addEther(for wallet: Wallet, network: Web3Network) throws {
+        let ether = ERC20Token(ether: true)
+        do {
+            try wallet.add(token: ether,
+                           network: network)
+        } catch let error {
+            throw error
+        }
+        self.userDefaultKeys.setEtherAdded(for: wallet)
+    }
+    
     public func addEther(for wallet: Wallet) throws {
         let ether = ERC20Token(ether: true)
         
-        for networkID in 1...100 {
+        for networkID in 1...99 {
```

Comment:
I'd do the network abstraction that's much like DB entity:
- id
- endpoint
- name
- bool flag built-in/custom
- EIP155 support

Simultaneously you will not need to use web3swift internal Infura provider, but always specify a provider in full